apple/internal/BUILD

```diff
@@ -22,3 +27,10 @@ filegroup(
         "//apple:__subpackages__",
     ],
 )
+
+environment_plist(
```

Comment:
we'll need to create 4 of these targets, once per platform, otherwise `macos_application` would use the default ios one since `platform_type` is not being set anywhere else. 

and then the attribute can be something like:

```
"_environment_plist": attr.label(
            default = "@build_bazel_rules_apple//apple/internal:environment_plist_{}".format(platform),
)
```

you'll need to extract this attribute from the `_COMMON_PRIVATE_TOOL_ATTRS` dictionary into  the `_create_apple_bundling_rule` and `_create_apple_binary_rule` methods. You can create a new method for common attributes that uses the platform.

Once we migrate to transitions, we'll be able to simplify this greatly :)